### ISSUE TITLE
Stack Overflow in rexex matching when using jdee-flycheck

### ISSUE THREAD

**Issue by oldButNotWise** [jdee-emacs/jdee#84]
I've got a rather large classpath, and this ends up causing problems with some of the error parsing code. (The line in the JDEE Compile Server buffer that sets up the classpath is over 61k characters!) 

There are a couple of changes that improve things:

In the function `jdee-compile-finish-kill-buffer`, there is no need for the ".*" patterns in the string-match tests.
```
(defun jdee-compile-finish-kill-buffer (buf msg)
  "Removes the jdee-compile window after a few seconds if no errors."
  (with-current-buffer buf
    (if (null (or (string-match "exited abnormally" msg)
		  (string-match "BUILD FAILED" (buffer-string))))
        ;; and so on
```

The regex used in `jdee-flycheck-compile-buffer-after` is more challenging. I replaced the call to `re-search-forward` with a call to this new function.
```
(defun jdee-flycheck-find-next-error ()
  ;; Complicated, to avoid stack overflow while executing regex search over possibly long lines
  (if (search-forward "error:" nil t)
      (progn
        (beginning-of-line)
        (or (re-search-forward "^\\(.*\\):\\([0-9]+\\): *error:\\(.*\\)$" (save-excursion (end-of-line) (point)) t)
            (progn (forward-line) (jdee-flycheck-find-next-error))))))
```

With those two changes I can now successfully run flycheck.

**Comment by m0smith** [comment]
Could you create a pull request with your changes?  That makes it much easier to track and include.

Is the project you are working with public?  If not can you share the dependency list? It would be great to have a large real-world project to test against.

**Comment by oldButNotWise** [comment]
Project is, alas, internal. I have posted a version of the prj file to http://pastebin.com/w7Rm3R50

A pull request is on the way.

**Comment by pwojnowski** [comment]
Closed in #86
